docs/design/job_architecture.md

```diff
@@ -278,133 +175,68 @@ These are all caught by the `JobCommChannel` on the browser side, then parsed an
 }
 ```
 
-For example:
-```json
-{
-  "msg_id": "some_string",
-  "data": {
-    "msg_type": "job_status",
-    "content": {
-      "example_job_id": {
-        "jobState": {
-          "job_id": "example_job_id",
-          "status": "running",
-          ... other state keys ...
-        },
-        "outputWidgetInfo": {}
-      }
-    }
-  }
-}
-```
-
-These are described below. The name (`msg_type`) is given, followed by the keys given in the `content` block.
+The full set of messages sent by the backend are described below. The name (`msg_type`) is given, followed by the keys given in the `content` block.
 
-By design, these should only be seen by the `JobCommChannel` instance, then sent into bus messages that get sent on specific channels. That information is also given in each block.
+The backend bundles together multiple messages of the same type in an object indexed by key (usually job or cell ID). In nearly all cases, the frontend then separates out the data and sends it out on individual channels for each job, batch job, or cell. The exception to this is `ERROR` messages; if the error contains data pertaining to the original request, the request parameters (`JOB_ID`, `BATCH_ID`, etc.) can be used to route the message to specific frontend components. Otherwise, the JobCommChannel emits the message in the browser console to assist in debugging.
 
-The backend bundles together multiple messages of the same type in an object indexed by key (usually job or cell ID). In nearly all cases, the frontend then separates out the data and sends it out on individual channels for each job or cell.
+By design, these messages are only seen by the `JobCommChannel`; the JobComm interface does not constitute an API for narrative job data.
 
 ### `ERROR`
-A general job comm error, capturing most errors that get thrown by the kernel
 
-**content** (this varies, but usually includes the below)
-  * `raw_request` - the original request message that wound up in an error
-  * `source` - request type or method that triggered the error
+A general job comm error message, capturing exceptions that get thrown by the kernel. The frontend uses the `request` data to distribute error messages on the appropriate job or cell channels.
+
+Content varies, but usually includes the below:
+  * `source` - what triggered the error; this may be the request type, arbitrary string, or null, meant to indicate what triggered the error
+  * `request` - request data, arbitrary string, or null, meant to indicate the comm request that triggered the error.
+

```

Comment:
This is still not clear, mostly because there are several different forms that backend errors can take. I've got to do some error handling stuff in the frontend as part of my devPoC work so I'm going to revisit this bit of documentation then.

Reply:
It should usually be the request type/data, because usually the error would result from a comm request

Reply:
I think once we get rid of the backend job update loop, the places where the `source` and `request` are a string will disappear, so that'll simplify things.

Reply:
amended this to mention the job lookup loop errors.